src/libraries/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/XmlKeyHelper.cs

```diff
@@ -264,24 +265,40 @@ internal bool HasElement(string localName)
 
             private static class Functions
             {
-                private static readonly Type s_xDocument = Type.GetType("System.Xml.Linq.XDocument, System.Private.Xml.Linq, Version=4.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51")!;
-                private static readonly Func<string, object> s_xDocumentCreate =
-                    s_xDocument.GetMethod(
+                private const string XmlLinqAssemblyString = ", System.Private.Xml.Linq, Version=4.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51";
+
+#pragma warning disable CA1810 // explicit static cctor
```

Comment:
An alternative approach considered was https://github.com/eerhardt/runtime/commit/90e8136db3c846a0af1129c2a711459fcadca98b. But that approach was disregarded because it involves linker suppressions.

Thoughts?

Reply:
I feel like I understand the alternative approach better (that is, I understand what the linker impact is, and I'm not sure why this is different than the existing approach).  

I'm probably fine with whichever you think is better... this class isn't going to change very much.  But if you're sticking with this way, please move the fields above the cctor :smile:.

Reply:
> and I'm not sure why this is different than the existing approach

The difference is that the ILLinker:

1. Treats readonly fields and local variables differently. See https://github.com/mono/linker/issues/1664. In the existing approach, it sets the `Type s_xDocument` field and then uses it later and the linker warns about it.
2. This statically references the `XElement` and `XName` types instead of walking through the PropertyType, which the linker isn't able to walk through. In this approach, the linker knows the local Type is `XElement`, and can see we are talking about the `Name` and `Value` properties.

> please move the fields above the cctor

done.